sql/event_data_objects.cc

```diff
@@ -446,18 +464,8 @@ Event_job_data::load_from_row(THD *thd, TABLE *table)
   Event_creation_ctx::load_from_db(thd, &mem_root, dbname.str, name.str, table,
                                    &creation_ctx);
 
-  ptr= strchr(definer.str, '@');
-
-  if (! ptr)
-    ptr= definer.str;
-
-  len= ptr - definer.str;
-  definer_user.str= strmake_root(&mem_root, definer.str, len);
```

Comment:
@ldonoso in commit message, you told 8.0 already uses strrchar instead of strchar()? :thinking: